.github/workflows/auto_branching.yml

```diff
@@ -0,0 +1,294 @@
+            echo "protection-outcome=failure" >> "$GITHUB_OUTPUT"
+            exit 1
+          fi
+
+  auto-branching-master:
+    name: master - raise PR with changes
+    runs-on: ubuntu-latest
+    needs: check-group-membership
+    if: ${{ needs.check-group-membership.outputs.member == 'true' }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Update target branch label in dependabot yml file
+        id: update-dependabot
+        run: |
+          # Read the dependabot.yml file
+          FILE_PATH="./.github/dependabot.yml"
+          TARGET_BRANCH="${{ github.event.inputs.target_branch }}"
+          # Append the target branch label to the labels node
+          awk -v target="'$TARGET_BRANCH'" '/^ *labels:/ {$0 = $0 "\n      - " target} 1' "$FILE_PATH" > temp.yml && mv temp.yml "$FILE_PATH"
+
+      - name: Remove lines with @pytest.mark.stream
+        id: remove-mark-stream
+        run: |
+          # Loop through files in the folder
+          grep -rl "tests/foreman" -e '@pytest\.mark\.stream' | while IFS= read -r file; do
+           awk '!/@pytest\.mark\.stream/' "$file" > temp && mv temp "$file"
+          done
+
+      - name: Update the Constants in __init__.py file
+        run: |
+          version="${{ github.event.inputs.target_branch }}"
+          ga_version="${{ github.event.inputs.ga_version }}"
+          old_stream_version="${version%.z}"
+          new_stream_version="${{ github.event.inputs.stream_version }}"
+          non_ga_versions="['$old_stream_version', '$new_stream_version']"
+          FILE_PATH="./robottelo/constants/__init__.py"
+          # update the version
+          sed -i.bak "s/SATELLITE_VERSION = \"$old_stream_version\"/SATELLITE_VERSION = \"$new_stream_version\"/" "$FILE_PATH"
+          sed -i.bak "s/  SATELLITE_VERSION: \"$old_stream_version\"/  SATELLITE_VERSION: \"$new_stream_version\"/" ./conf/robottelo.yaml.template
+          sed -i.bak "s/SAT_NON_GA_VERSIONS = \[.*\]/SAT_NON_GA_VERSIONS = $non_ga_versions/" "$FILE_PATH"
+          rm "$FILE_PATH.bak" "./conf/robottelo.yaml.template.bak"
+
+      - name: git status
+        run: git status
+
+      - name: git diff
+        run: git diff
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          branch_name="auto-branching-${{ github.event.inputs.target_branch }}-$(date '+%s')"
+          git checkout -b "$branch_name"
+          git add setup.py ./tests/foreman ./robottelo/* ./requirements.txt ./.github/* ./conf/robottelo.yaml.template
+          git commit -m "Changes for ${{ github.event.inputs.target_branch }} new branch"
+          git remote -vvv
+          git push origin "$branch_name"
+
+      - name: Create pull request
+        id: create_pr
+        run: |
+          title="[master]: Changes for ${{ github.event.inputs.target_branch }} new branch"
```

Comment:
```suggestion
          title="[${{ github.event.inputs.target_branch }}]: Changes for new ${{ github.event.inputs.target_branch }} branch"
```

Reply:
github.event.inputs.target_branch is point to 6.16